src/order.rs

```diff
@@ -456,6 +456,28 @@ impl ChallengeHandle<'_> {
         }
     }
 
+    /// Notify the server that the challenge is ready and provide a payload
+    ///
+    /// This function is for the ACME challenge device-attest-01.
+    /// See <https://datatracker.ietf.org/doc/draft-acme-device-attest/> for details.
+    ///
+    /// Note: Do not use this function for http-01, tls-alpn-01 or dns-01 challenges.
+    ///
+    /// `payload` should serialize to "attObj" as defined in link.
+    pub async fn set_ready_with_payload(&mut self, payload: &impl Serialize) -> Result<(), Error> {
```

Comment:
Should we add a `ChallengeType` and error if the `ChallengeType` is not the expected one?

I suggest we take a more specific type here, maybe:

```rust
#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
struct DeviceAttestation<T> {
     att_obj: T,
}
```

and then take `object: &DeviceAttestation<impl Serialize>`?

Maybe rename to `send_device_attestation()`?